pipelined/bevy_sprite2/src/render/mod.rs

```diff
@@ -180,14 +358,50 @@ pub fn extract_atlases(
         }
     }
 
-    if let Some(mut extracted_sprites_res) = render_world.get_resource_mut::<ExtractedSprites>() {
+    if let Some(mut extracted_sprites_res) =
+        render_world.get_resource_mut::<ExtractedSprites<Sprite3d>>()
+    {
+        extracted_sprites_res.sprites.extend(extracted_sprites);
+    }
+}
+
+pub fn extract_sprites_2d(
+    images: Res<Assets<Image>>,
+    sprite_query: Query<(&Sprite2d, &GlobalTransform, &Handle<Image>)>,
+    mut render_world: ResMut<RenderWorld>,
+) {
+    let mut extracted_sprites = Vec::new();
+    for (sprite, transform, handle) in sprite_query.iter() {
+        let image = if let Some(image) = images.get(handle) {
+            image
+        } else {
+            continue;
+        };
+        let size = image.texture_descriptor.size;
+
+        extracted_sprites.push(ExtractedSprite {
+            atlas_size: None,
+            transform: transform.compute_matrix(),
+            rect: Rect {
+                min: Vec2::ZERO,
+                max: sprite
+                    .custom_size
+                    .unwrap_or_else(|| Vec2::new(size.width as f32, size.height as f32)),
+            },
+            handle: handle.clone_weak(),
+        });
+    }
+
+    if let Some(mut extracted_sprites_res) =
+        render_world.get_resource_mut::<ExtractedSprites<Sprite2d>>()
+    {
         extracted_sprites_res.sprites.extend(extracted_sprites);
     }
 }
 
-pub fn extract_sprites(
+pub fn extract_sprites_3d(
```

Comment:
I wonder if this code really needs to be duplicated, for `*_2d` and `*_3d`? The bodies of the functions seem identical, except for the `Sprite2d`/`Sprite3d` types? Could/should this be made generic?

Reply:
Say this would be generic. Now we have `extract_sprites<T>`. We query for `&T`. How do we access `custom_size`, `flip_x`or `flip_y` of any type `T`? So we have to implement a trait `Sprite`. I just didn't find it worth the effort to generalize this if it probably diverges in the future as elaborated above.